src/main/java/org/hawkular/client/android/auth/SecretStoreAuthzModule.java

```diff
@@ -0,0 +1,185 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.client.android.auth;
+
+import java.net.URI;
+import java.net.URL;
+import java.util.UUID;
+
+import org.jboss.aerogear.android.authorization.AuthzModule;
+import org.jboss.aerogear.android.authorization.oauth2.OAuth2AuthzSession;
+import org.jboss.aerogear.android.core.Callback;
+import org.jboss.aerogear.android.pipe.http.HeaderAndBody;
+import org.jboss.aerogear.android.pipe.http.HttpException;
+import org.jboss.aerogear.android.pipe.http.HttpProvider;
+import org.jboss.aerogear.android.pipe.http.HttpRestProvider;
+import org.jboss.aerogear.android.pipe.module.AuthorizationFields;
+import org.jboss.aerogear.android.pipe.module.ModuleFields;
+import org.jboss.aerogear.android.store.DataManager;
+import org.jboss.aerogear.android.store.generator.IdGenerator;
+import org.jboss.aerogear.android.store.sql.SQLStore;
+import org.jboss.aerogear.android.store.sql.SQLStoreConfiguration;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.util.Base64;
+import timber.log.Timber;
+
+/**
+ * ModifiedAuthzModule.
+ * <p/>
+ * Performs all related work of generating new key and secret pair, Storing them as account,
+ * Retrieving stored account, Deleting stored account.
+ */
+
+public class SecretStoreAuthzModule implements AuthzModule {
+
+    private Context context;
+    private SQLStore<OAuth2AuthzSession> sessionStore;
+
+    public SecretStoreAuthzModule(Context context) {
+        this.context = context.getApplicationContext();
+        sessionStore = openSessionStore();
+    }
+
+    private SQLStore<OAuth2AuthzSession> openSessionStore() {
+        DataManager.config(AuthEndPoints.STORE, SQLStoreConfiguration.class)
+                .forClass(OAuth2AuthzSession.class)
+                .withContext(context)
+                .withIdGenerator(new IdGenerator() {
+                    @Override
+                    public String generate() {
+                        return UUID.randomUUID().toString();
+                    }
+                }).store();
+        sessionStore = (SQLStore<OAuth2AuthzSession>) DataManager.getStore(AuthEndPoints.STORE);
+        sessionStore.openSync();
+        return sessionStore;
+    }
+
+    @Override
+    public boolean isAuthorized() {
+        return true;
+    }
+
+    @Override
+    public boolean hasCredentials() {
+        OAuth2AuthzSession storedAccount = sessionStore.read(AuthEndPoints.NAME);
+        return storedAccount != null;
+    }
+
+    public void requestAccess(final Activity activity, final Callback<String> callback,
+                              final String username, final String password, final URL url) {
+
+        Exception exception;
```

Comment:
I’m pretty sure that these two variables are useless, because they are duplicated in the inner class.